services/src/tests/e2e/tests/authorization_with_queries.spec.ts

```diff
@@ -0,0 +1,211 @@
+    type Employee {
+      id: String
+      name: String
+      hireDate: Int
+      department: Department!
+      address: String @policy(policies: [{
+        namespace: "namespace",
+        name: "notClassified",
+        args: {
+          departmentId: "{source.department.id}",
+          hireDate: "{source.hireDate}"
+        }
+      }])
+    }
+
+    type Query {
+      allowedEmployee: Employee! @stub(value: {
+        id: "1"
+        name: "John Smith"
+        address: "Tel Aviv"
+        hireDate: 2010
+        department: {
+          id: "D1"
+          name: "Sales"
+        }
+      })
+      deniedEmployee1: Employee! @stub(value: {
+        id: "2"
+        name: "Mark Zuckerberg",
+        department: {
+          id: "D1000"
+          name: "VIP"
+        }
+        hireDate: 2010
+        address: "Facebook HQ"
+      })
+      deniedEmployee2: Employee! @stub(value: {
+        id: "2"
+        name: "Tom Baker",
+        department: {
+          id: "D2"
+          name: "VIP"
+        }
+        hireDate: 2019
+        address: "Atlanta"
+      })
+      classifiedDepartments: [Department!]! @stub(value: [{
+        id: "D1000"
+        name: "VIP"
+      }])
+    }
+  `,
+};
+
+const createPolicyMutation = `
+  mutation CreatePolicy($policies: [PolicyInput!]!) {
+    updatePolicies(input: $policies) {
+      success
+    }
+  }`;
+
+const employeeQuery = (query: string) => `
+  query {
+    ${query} {
+      id
+      name
+      address
+    }
+  }
+`;
+
+interface CreatePolicyMutationResponse {
+    updatePolicies: {
+        success: boolean;
+    };
+}
+
+interface UpdateSchemasMutationResponse {
+    updateSchemas: {
+        success: boolean;
+    };
+}
+
+interface AllowedEmployeeQueryResponse {
+    allowedEmployee: {
+        id: string;
+        name: string;
+        address: string;
+    };
+}
+
+describe('Authorization with queries', () => {
+    let gatewayClient: GraphQLClient;
+    let registryClient: GraphQLClient;
+
+    beforeAll(() => {
+        gatewayClient = new GraphQLClient('http://localhost:8080/graphql');
+        registryClient = new GraphQLClient('http://localhost:8090/graphql');
+    });
+
+    test('Setup policies', async () => {
+        const policyResponse: CreatePolicyMutationResponse = await registryClient.request(createPolicyMutation, {
+            policies,
+        });
+        expect(policyResponse.updatePolicies.success).toBeTruthy();
+
+        const schemaResponse: UpdateSchemasMutationResponse = await registryClient.request(createSchemaMutation, {
+            schema,
+        });
+        expect(schemaResponse.updateSchemas.success).toBeTruthy();
+
+        // Wait for gateway to update before next tests
+        await sleep(500);
+    });
+
+    test('Query allowed employee', async () => {
+        const response: AllowedEmployeeQueryResponse = await gatewayClient.request(employeeQuery('allowedEmployee'));
+        expect(response.allowedEmployee.address).toBeDefined();
```

Comment:
Why not check the actual value of the address? should be pretty simple here

Reply:
I'll add snapshot validation